pkg/kubectl/cmd/get.go

```diff
@@ -219,33 +219,55 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 		return err
 	}
 
-	allErrs := []error{}
-	infos, err := r.Infos()
-	if err != nil {
-		allErrs = append(allErrs, err)
-	}
-
 	if generic {
 		clientConfig, err := f.ClientConfig()
 		if err != nil {
 			return err
 		}
 
-		singular := false
-		r.IntoSingular(&singular)
-
 		// the outermost object will be converted to the output-version, but inner
 		// objects can use their mappings
 		version, err := cmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
 		if err != nil {
 			return err
 		}
-		obj, err := resource.AsVersionedObject(infos, !singular, version, f.JSONEncoder())
-		if err != nil {
+
+		switch options.Recursive {
```

Comment:
Recursive should still output a List, not individual items.  So this seems incorrect in that it will result in many lists printed.